### ISSUE TITLE
Object Tracking and Re-ID

### ISSUE THREAD

**Issue by changtimwu** [changtimwu/changtimwu.github.com#103]
ReID implementation on Jetson https://github.com/anonymous1computervision/Human-ReID-in-NVIDIA-Jetson

**Comment by changtimwu** [comment]
* [Deep Learning for Person Re-identification: A Survey and Outlook](https://arxiv.org/pdf/2001.04193.pdf)
* OmniScale network: [paper](https://arxiv.org/abs/2003.07618), [implementation](https://github.com/opencv/openvino_training_extensions)
* [TorchREID doc](https://kaiyangzhou.github.io/deep-person-reid/index.html)
* MGN: [paper](https://arxiv.org/abs/1804.01438v3), [paper](https://github.com/levyfan/reid-mgn)
* BFENet: [Intro](https://zhuanlan.zhihu.com/p/53241545), [paper](https://arxiv.org/abs/1811.07130)

**Comment by changtimwu** [comment]
## datasets
* http://robustsystems.coe.neu.edu/sites/robustsystems.coe.neu.edu/files/systems/projectpages/reiddataset.html

**Comment by changtimwu** [comment]
## Pytorch
* https://pytorch.org/tutorials/beginner/pytorch_with_examples.html

**Comment by changtimwu** [comment]
## the domain shift problem in ReID

_Many works are devoted to the cross-domain adaptation problem [6], [10], [11] because this question is critical for practical use of reidentification models._

* [6][In Defense of the Triplet Loss Again: Learning Robust Person Re-Identification with Fast Approximated Triplet Loss and Label Distillation](https://arxiv.org/abs/1912.07863)
_The comparative losses (typically, triplet loss) are appealing choices for learning person re-identification (ReID) features. However, the triplet loss is computationally much more expensive than the (practically more popular) classification loss, limiting their wider usage in massive datasets. Moreover, the abundance of label noise and outliers in ReID datasets may also put the margin-based loss in jeopardy. This work addresses the above two shortcomings of triplet loss, extending its effectiveness to large-scale ReID datasets with potentially noisy labels. We propose a fast-approximated triplet (FAT) loss, which provably converts the point-wise triplet loss into its upper bound form, consisting of a point-to-set loss term plus cluster compactness regularization. It preserves the effectiveness of triplet loss, while leading to linear complexity to the training set size. A label distillation strategy is further designed to learn refined soft-labels in place of the potentially noisy labels, from only an identified subset of confident examples, through teacher-student networks. We conduct extensive experiments on three most popular ReID benchmarks (Market-1501, DukeMTMC-reID, and MSMT17), and demonstrate that FAT loss with distilled labels lead to ReID features with remarkable accuracy, efficiency, robustness, and direct transferability to unseen datasets._
* [10][Fairest of Them All: Establishing a Strong Baseline for Cross-Domain Person ReID](https://arxiv.org/abs/1907.12016)
_Person re-identification (ReID) remains a very difficult challenge in computer vision, and critical for large-scale video surveillance scenarios where an individual could appear in different camera views at different times. There has been recent interest in tackling this challenge using cross-domain approaches, which leverages data from source domains that are different than the target domain. Such approaches are more practical for real-world widespread deployment given that they don't require on-site training (as with unsupervised or domain transfer approaches) or on-site manual annotation and training (as with supervised approaches). In this study, we take a systematic approach to establishing a large baseline source domain and target domain for cross-domain person ReID. We accomplish this by conducting a comprehensive analysis to study the similarities between source domains proposed in literature, and studying the effects of incrementally increasing the size of the source domain. This allows us to establish a balanced source domain and target domain split that promotes variety in both source and target domains. Furthermore, using lessons learned from the state-of-the-art supervised person re-identification methods, we establish a strong baseline method for cross-domain person ReID. Experiments show that a source domain composed of two of the largest person ReID domains (SYSU and MSMT) performs well across six commonly-used target domains. Furthermore, we show that, surprisingly, two of the recent commonly-used domains (PRID and GRID) have too few query images to provide meaningful insights. As such, based on our findings, we propose the following balanced baseline for cross-domain person ReID consisting of: i) a fixed multi-source domain consisting of SYSU, MSMT, Airport and 3DPeS, and ii) a multi-target domain consisting of Market-1501, DukeMTMC-reID, CUHK03, PRID, GRID and VIPeR._
* [11][Calibrated Domain-Invariant Learning for Highly Generalizable Large Scale Re-Identification](https://arxiv.org/abs/1911.11314)
_Many real-world applications, such as city-scale traffic monitoring and control, requires large-scale re-identification. However, previous ReID methods often failed to address two limitations in existing ReID benchmarks, i.e., low spatiotemporal coverage and sample imbalance. Notwithstanding their demonstrated success in every single benchmark, they have difficulties in generalizing to unseen environments. As a result, these methods are less applicable in a large-scale setting due to poor generalization. In seek for a highly generalizable large-scale ReID method, we present an adversarial domain invariant feature learning framework (ADIN) that explicitly learns to separate identity-related features from challenging variations, where for the first time "free" annotations in ReID data such as video timestamp and camera index are utilized. Furthermore, we find that the imbalance of nuisance classes jeopardizes the adversarial training, and for mitigation we propose a calibrated adversarial loss that is attentive to nuisance distribution. Experiments on existing large-scale person vehicle ReID datasets demonstrate that ADIN learns more robust and generalizable representations, as evidenced by its outstanding direct transfer performance across datasets, which is a criterion that can better measure the generalizability of large-scale ReID methods/_

**Comment by changtimwu** [comment]
GhostNet: https://zhuanlan.zhihu.com/p/109325275
_our GhostNet can achieve higher recognition performance (e.g. 75.7% top-1 accuracy) than MobileNetV3 with similar computational cost on the ImageNet ILSVRC-2012 classification dataset._

**Comment by changtimwu** [comment]
* [26] K. Zhou, Y. Yang, A. Cavallaro, and T. Xiang, [Omni-scale feature learning for person reidentification](https://arxiv.org/abs/1905.00953) in ICCV, 2019.
* [27] E. Izutov, “Fast and accurate person re-identification with rmnet,” 2018
* [34]  K. Zhou, Y. Yang, A. Cavallaro, and T. Xiang, [Learning generalisable omni-scale representations for person re-identification](https://arxiv.org/abs/1910.06827), [code](https://github.com/KaiyangZhou/deep-person-reid)

**Comment by changtimwu** [comment]
* [VisDA2020: 4th Visual Domain Adaptation Challenge](https://zhuanlan.zhihu.com/p/137644578)
* [Dissecting Person Re-identification from the Viewpoint of Viewpoint](https://arxiv.org/abs/1812.02162)

**Comment by changtimwu** [comment]
* consider point of view: https://arxiv.org/pdf/1812.02162.pdf
* https://github.com/NEU-Gou/awesome-reid-dataset

**Comment by changtimwu** [comment]
# tracking is the point
* brief introduction  https://zhuanlan.zhihu.com/p/97449724

## MOTS 
* https://motchallenge.net/results/CVPR_2020_MOTS_Challenge/
* https://paperswithcode.com/paper/mot20-a-benchmark-for-multi-object-tracking

## KLT tracker
* https://en.wikipedia.org/wiki/Kanade%E2%80%93Lucas%E2%80%93Tomasi_feature_tracker
* GPU implementation https://www.diva-portal.org/smash/get/diva2:241567/FULLTEXT01.pdf

**Comment by changtimwu** [comment]
good survey
https://zhuanlan.zhihu.com/p/95499141

**Comment by changtimwu** [comment]
# Object Tracking
excellent research & implementation on video based object tracking

https://mp.weixin.qq.com/s?__biz=MzIwMTE1NjQxMQ==&mid=2247497851&idx=3&sn=a99fbb60d9102df7e74f84b4559817a7&chksm=96f08a2fa187033909ae79b37163fb099e75482946fd0be0475c5121a702d38aaad70d369f4e

https://github.com/MegviiDetection/video_analyst

* [SiamFC++: Towards Robust and Accurate Visual Tracking with Target Estimation Guidelines](https://arxiv.org/abs/1911.06188)
* [State-Aware Tracker for Real-Time Video Object Segmentation](https://arxiv.org/abs/2003.00482)

[Solid multi object tracking class](https://www.youtube.com/watch?v=CsYIlsGLqg4)


original siam network: https://www.youtube.com/watch?v=jZoUalMMZ_0

**Comment by changtimwu** [comment]
## 3D reconstruction
* https://bangqu.com/749q45.html
